clients/executor-http-client/src/lib.rs

```diff
@@ -17,6 +17,7 @@ pub enum Error {
 pub struct ExecutorHttpClient {
     client: Client,
     executor_address: Url,
+    auth: Option<(String, String)>, // Optional Basic Auth credentials
```

Comment:
While probably expected, maybe it's worth considering wrapping it in a `PasswordAuth` struct or similar to avoid confusions.

Reply:
Updated, both options seems okay, although nomos-cli and executor-http-client will probably change a lot in the future.